### PR TITLE
new na_ontap_motd module

### PR DESCRIPTION
**Pull request by dprts**
##### SUMMARY
This module allows setting/removing cluster and SVM level MOTD 

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
- na_ontap_motd
##### ANSIBLE VERSION
2.7

### CODE REVIEW

**Review by gundalow — CHANGES_REQUESTED** [review]
Needs review by someone in the NetApp org

File: lib/ansible/modules/storage/netapp/na_ontap_motd.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    - This module allows you to manipulate motd on cDOT
+version_added: "2.7"
+requirements:
+    - netapp_lib
+options:
+    state:
+        description:
+        - If C(state=present) sets MOTD given in I(message) C(state=absent) removes it.
+        required: true
+        choices: ['present', 'absent']
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_motd.py]: I would make present the default

File: lib/ansible/modules/storage/netapp/na_ontap_motd.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        ))
+
+        self.module = AnsibleModule(
+            argument_spec=argument_spec,
+            supports_check_mode=True
+        )
+
+        params = self.module.params
+
+        self.vserver = params['vserver']
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_motd.py]: only FYI.
We found this not to be very useful, and we now prefer to assign the whole params dictionary to a single variable.

File: lib/ansible/modules/storage/netapp/na_ontap_motd.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        if HAS_NETAPP_LIB is False:
+            self.module.fail_json(msg="the python NetApp-Lib module is required")
+
+        self.server = netapp_utils.setup_na_ontap_zapi(module=self.module)
+
+    def _create_call(self):
+        api_call = netapp_utils.zapi.NaElement('vserver-motd-modify-iter')
+        api_call.add_new_child('message', self.message)
+        api_call.add_new_child('is-cluster-message-enabled', 'true' if self.show_cluster_motd else 'false')
+        query = netapp_utils.zapi.NaElement('query')
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_motd.py]: I don't know if the query element is really needed.  It seems superfluous as the API is already directed to this vserver.   (But we found other cases where the request seems over qualified).

**Review by carchi8py — APPROVED** [review]

**Comment by carchi8py** [comment]
@gundalow we are good with this one being merged in.

**Review by lonico — APPROVED** [review]

File: lib/ansible/modules/storage/netapp/na_ontap_motd.py (hunk 1 above)
Comment (dprts) [reply lib/ansible/modules/storage/netapp/na_ontap_motd.py]: Agree, I am going to correct it.

File: lib/ansible/modules/storage/netapp/na_ontap_motd.py (hunk 2 above)
Comment (dprts) [reply lib/ansible/modules/storage/netapp/na_ontap_motd.py]: Also I am going to apply this change

File: lib/ansible/modules/storage/netapp/na_ontap_motd.py (hunk 3 above)
Comment (dprts) [reply lib/ansible/modules/storage/netapp/na_ontap_motd.py]: It looks like here it is necessary but i will check it once more

**Comment by dprts** [comment]
By mistake I pushed rebased branch. I am closing this and open new PR.

**Pull request closed without merging**